tmt/steps/__init__.py

```diff
@@ -779,10 +781,21 @@ def _patch_raw_datum(
                 else:
                     debug('  incompatible step data')
 
-                    raw_datum = {
-                        'name': raw_datum['name'],
-                        'how': how
-                        }
+                    data_base = self._plugin_base_class._data_class
+
+                    debug('  compatible base', f'{data_base.__module__}.{data_base.__name__}')
+                    debug('  compatible keys', list(data_base.keys()))
+
+                    # Copy compatible keys only, ignore everything else
+                    # SIM118: Use `{key} in {dict}` instead of `{key} in {dict}.keys()`.
```

Comment:
@happz why we cannot use `key in data_base` ? it is not clear to the reader.

Reply:
Because there's no `data_base.__iter__()`. ruff tries to drop `keys()` on the assumption that if `data_base` is dict-like, and has `keys()`, it's iterabel directly, i.e. `for key in data_base` should be more pythonic variant of `for key in data_base.keys()`. Which is not true, `keys()` is the only way with `data_base`.

Reply:
Added few more words in 0699c2fd.

WRT adding `__iter__()` - yes, that's the plan.

Reply:
Seems to be addressed.